raiden/messages.py

```diff
@@ -1700,6 +1704,112 @@ def verify_request_monitoring(
         )
 
 
+class UpdatePFS(SignedMessage):
+    cmdid = messages.UPDATEPFS
+    """ Message to inform a pathfinding service about a capacity change. """
+
+    def __init__(
+            self,
+            nonce: typing.Nonce,
+            transferred_amount: typing.TokenAmount,
+            locked_amount: typing.TokenAmount,
+            locksroot: typing.Locksroot,
+            token_network_address: typing.TokenNetworkAddress,
+            channel_identifier: typing.ChannelID,
+            chain_id: typing.ChainID,
+            reveal_timeout: int,
+            signature: typing.Optional[typing.Signature] = None,
+    ) -> None:
+        self.chain_id = chain_id
+        self.nonce = nonce
+        self.transferred_amount = transferred_amount
+        self.locked_amount = locked_amount
+        self.locksroot = locksroot
+        self.channel_identifier = channel_identifier
+        self.token_network_address = token_network_address
+        self.reveal_timeout = reveal_timeout
+        if signature is None:
+            self.signature = b''
+        else:
+            self.signature = signature
+
+    @classmethod
+    def from_balance_proof(
+            cls: typing.Type[typing.T_UpdatePFS],
+            balance_proof: BalanceProofUnsignedState,
+            reveal_timeout: int,
+    ) -> typing.T_UpdatePFS:
+        assert isinstance(balance_proof, BalanceProofUnsignedState)
+        return cls(
+            nonce=balance_proof.nonce,
+            transferred_amount=balance_proof.transferred_amount,
+            locked_amount=balance_proof.locked_amount,
+            locksroot=balance_proof.locksroot,
+            token_network_address=balance_proof.token_network_identifier,
+            channel_identifier=balance_proof.channel_identifier,
+            chain_id=balance_proof.chain_id,
+            reveal_timeout=reveal_timeout,
+        )
+
+    def to_dict(self) -> typing.Dict:
+        return {
+            'type': self.__class__.__name__,
+            'chain_id': self.chain_id,
+            'nonce': self.nonce,
+            'token_network_address': to_normalized_address(self.token_network_address),
+            'channel_identifier': self.channel_identifier,
+            'transferred_amount': self.transferred_amount,
+            'locked_amount': self.locked_amount,
+            'locksroot': encode_hex(self.locksroot),
+            'signature': encode_hex(self.signature),
+            'reveal_timeout': self.reveal_timeout,
+        }
+
+    @classmethod
+    def from_dict(
+            cls: typing.Type[typing.T_UpdatePFS],
+            data: typing.Dict,
+    ) -> typing.T_UpdatePFS:
+        return cls(
+            nonce=data['nonce'],
+            transferred_amount=data['transferred_amount'],
+            locked_amount=data['locked_amount'],
+            locksroot=data['locksroot'],
+            token_network_address=data['token_network_identifier'],
```

Comment:
Same question as above

Reply:
this one is a bug! thanks for spotting it!